### PR TITLE
add test_acs_full_sync.py

### PR DESCRIPTION
**Pull request by jinzhaochangAtMicrosoft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py
Comment (shpengmsft) [inline sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py:1]: read through the contents in this _client.py file, should we move it to azure-ai-generative?

File: sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import re
+from typing import Optional
+
+from azureml._base_sdk_common.service_discovery import get_service_url
+from azureml.core import Run, Workspace
+from azureml.exceptions import UserErrorException
+from azure.ai.generative.index._utils.logging import get_logger
```
Comment (shpengmsft) [inline sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py]: we shouldn't refer to azure-ai-generative from ai-resources. that's another reason i think maybe we should move it to generative.

File: sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py
Comment (jinzhaochangAtMicrosoft) [reply sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py:1]: I would rather leave it there for completeness and put out another PR to deal with the duplicates and deprecations in the future.

File: sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py (hunk 1 above)
Comment (jinzhaochangAtMicrosoft) [reply sdk/ai/azure-ai-resources/azure/ai/resources/_asset_client/_client.py]: My bad. azure-ai-resources has its own logging as well.

**Review by shpengmsft — APPROVED** [review]

**Pull request merged**